framework/include/outputs/PetscOutput.h

```diff
@@ -101,4 +121,6 @@ class PetscOutput : public Output
 
   /// Linear residual output end time
   Real _linear_end_time;
+
+  friend class PetscOutputInterface;
```

Comment:
why do you need to be friend with a class you inherit

Reply:
Other direction